uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -977,30 +971,25 @@ define([
       }
 
       /**
-       * Loads just one pre-selected managee
-       *
-       * @return {Promise}
-       */
-      function loadManagee () {
-        return Contact.find(this.directiveOptions.selectedContactId)
-          .then(function (contact) {
-            this.managedContacts = [contact];
-          }.bind(this));
-      }
-
-      /**
        * Loads the managees of currently logged in user
        *
        * @return {Promise}
        */
       function loadManagees () {
-        return Contact.find(this.directiveOptions.contactId)
-          .then(function (contact) {
-            return contact.leaveManagees();
-          })
-          .then(function (contacts) {
-            this.managedContacts = contacts;
-          }.bind(this));
+        if (this.directiveOptions.selectedContactId) {
```

Comment:
Pardon the one liner
```js
function loadManagees () {
  return Contact.find(this.directiveOptions.contactId)
    .then(function (contact) {
      return contact.leaveManagees();
    })
    .then(function (contacts) {
      this.managedContacts = this.directiveOptions.selectedContactId ? _.filter(contacts, function (contact) {
        return // match the contact id with `selectedContactId`
      }) : contacts;
    }.bind(this));
}
```

the logic being

1) get the current contact
2) fetch the managees of the contact
3) if we are specifying a specific managee, then find it in the list of managees
4) otherwise, use the full list